src/cobald/decorator/logger.py

```diff
@@ -10,6 +12,34 @@
 )
 
 
+class _WarnValue(NamedTuple):
+    value: Any
+    warning: Warning
+
+
+class _WarnMap(dict):
+    r"""Map raising ``warnings`` for specific keys pointing to ``_WarnValue``\ s"""
+
+    def __getitem__(self, item):
+        value = super().__getitem__(item)
+        if isinstance(value, _WarnValue):
+            warnings.warn(value.warning, stacklevel=2)
+            return value.value
+        else:
+            return value
+
+
+_DEPRECATION_MAP = _WarnMap(
+    value=10.0,
+    demand=10.0,
+    supply=10.0,
+    utilisation=0.5,
+    allocation=0.5,
+    consumption=_WarnValue(0.5, FutureWarning("")),
```

Comment:
Given the docstring of `Logger` I suppose that `consumption` is an alias for `allocation`. This currently is not clear from this mapping and how (as well as where) is this actually applied?